src/components/elements/favorites-list.tsx

```diff
@@ -0,0 +1,103 @@
+"use client";
+
+import useFavorites from "@lib/hooks/useFavorites";
+import { ChatBubbleLeftEllipsisIcon, ClipboardDocumentIcon, EnvelopeIcon, HeartIcon} from "@heroicons/react/24/outline";
+import {useIsClient} from "usehooks-ts";
+import { XMarkIcon } from "@heroicons/react/20/solid";
+
+
+const ShareButtons = () => {
+  const baseUrl = window.location.origin;
+  const { favs } = useFavorites();
+  const urlParams = `/favorites?fav=${favs.map(fav => `${fav.uuid}:${fav.title}:${fav.units}`).join(",")}`;
+
+  const handleCopy = async () => {
+    try {
+      await navigator.clipboard.writeText(baseUrl + urlParams);
+    } catch (error) {
+      throw new Error(`Failed to copy URL: ${error}`);
+    }
+  };
+
+  const handleSendEmail = () => {
+    const subject = "Check out these summer courses!";
+    const body = `Here is the link: ${baseUrl + urlParams}`;
+    window.location.href = `mailto:?subject=${encodeURIComponent(subject)}&body=${encodeURIComponent(body)}`;
```

Comment:
We should match the current link behavior on the site:
`mailto:?body=I saved some Stanford Summer Session courses that looked interesting to me. What do you think?%0D%0Ahttps://summer.stanford.edu/courses/favorites/3840&subject=Someone wants you to see their list of favorite courses from Stanford Summer Session`